### PR TITLE
(Wallet) Connect unlock action from WebUI to native unlock screen

### PR DESCRIPTION
**Pull request by simoarpe**
## Resolves https://github.com/brave/brave-browser/issues/40224

When Wallet is locked, show a new page that will open native unlock screen when double tapped.

https://github.com/user-attachments/assets/398d81d0-fc32-4d22-b77e-9bb866456dc7

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Navigate to Wallet section
- Type password to unlock the screen
- Tap on the Wallet icon in the address bar, tap on the overflow menu and select "Lock Wallet"
- Obsererve the WebUI page changes to a locked state asking to double tap
- Double tap on the screen
- Observe the native screen to unlock pops up
- Type the password again
- Observe the Wallet web page is visible

### CODE REVIEW

**Review by thypon — APPROVED** [review]

**Review by mkarolin — APPROVED** [review]
`strings`++

File: components/brave_wallet_ui/components/desktop/lock-screen/index.tsx [hunk 1]
```diff
@@ -99,6 +102,39 @@ export const LockScreen = () => {
...
+
+    // Methods
+    const onDoubleTap = () => {
+      const time = new Date().getTime()
+      const delta = time - lastPress
+      if (delta < 400) {
+        getWalletPageApiProxy().pageHandler.unlockWalletUI()
+      }
+      lastPress = time
+    }
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/lock-screen/index.tsx:117]: lets use the event detail to check if this was a double-click to better handle user accessibility: https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event#usage_notes

File: components/brave_wallet_ui/components/desktop/lock-screen/index.tsx [hunk 2]
```diff
@@ -99,6 +102,39 @@ export const LockScreen = () => {
...
+  if (isAndroid) {
+    // Computed
+    let lastPress = 0
+
+    // Methods
+    const onDoubleTap = () => {
+      const time = new Date().getTime()
+      const delta = time - lastPress
+      if (delta < 400) {
+        getWalletPageApiProxy().pageHandler.unlockWalletUI()
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/lock-screen/index.tsx:114]: Maybe use the generic proxy getter?
```suggestion
        getWalletApiProxy().pageHandler?.unlockWalletUI()
```

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/lock-screen/index.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/lock-screen/index.tsx]: Updated!

File: components/brave_wallet_ui/components/desktop/lock-screen/index.tsx [hunk 3]
```diff
@@ -99,6 +102,33 @@ export const LockScreen = () => {
 
   const isAndroid = loadTimeData.getBoolean('isAndroid') || false
 
+  if (isAndroid) {
+    // Methods
+    const onDoubleTap = (e: React.MouseEvent<HTMLButtonElement>) => {
+      if (e.detail === 2) {
+        getWalletAPIProxy().pageHandler?.unlockWalletUI()
+      }
+    }
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/lock-screen/index.tsx:111]: nit: can we move this to the existing `// methods` section?

File: components/brave_wallet_ui/components/desktop/lock-screen/index.tsx [hunk 4]
```diff
@@ -99,6 +102,33 @@ export const LockScreen = () => {
...
+        </Text>
+        <Text
+          textSize='16px'
+          textColor='tertiary'
+        >
+          {getLocale('braveWalletUnlockAndroidDescription')}
+        </Text>
+      </AndroidLockScreenWrapper>
+    )
+  }
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/lock-screen/index.tsx:130]: nit: Can we move this under the `// render` section?

**Review by josheleonard — APPROVED** [review]
Approved with comments

File: components/brave_wallet_ui/components/desktop/lock-screen/index.tsx (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/lock-screen/index.tsx]: Moved in https://github.com/brave/brave-core/pull/24992/commits/7bb8853179aa94459eddd166de6aecfc376cb0ce

File: components/brave_wallet_ui/components/desktop/lock-screen/index.tsx (hunk 4 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/lock-screen/index.tsx]: Moved in https://github.com/brave/brave-core/pull/24992/commits/7bb8853179aa94459eddd166de6aecfc376cb0ce

File: components/brave_wallet_ui/components/desktop/lock-screen/index.tsx [hunk 5]
```diff
@@ -97,9 +100,35 @@ export const LockScreen = () => {
     }
   }, [history, isPanel])
 
+  const onDoubleTap = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if (e.detail === 2) {
```
Comment (supermassive) [inline components/brave_wallet_ui/components/desktop/lock-screen/index.tsx:104]: needs brief comment on what is `2`

File: browser/ui/webui/brave_wallet/page_handler/wallet_page_handler.cc [hunk 6]
```diff
@@ -31,3 +31,7 @@ void WalletPageHandler::ShowApprovePanelUI() {
 void WalletPageHandler::ShowWalletBackupUI() {
   NOTREACHED_IN_MIGRATION();
 }
+
+void WalletPageHandler::UnlockWalletUI() {
+  NOTREACHED_IN_MIGRATION();
```
Comment (supermassive) [inline browser/ui/webui/brave_wallet/page_handler/wallet_page_handler.cc:36]: New code should use `NOTREACHED()`. 
It's ok to replace NOTREACHED_IN_MIGRATION()->NOTREACHED() for consistency.

File: browser/brave_wallet/brave_wallet_provider_delegate_impl_helper.cc [hunk 7]
```diff
@@ -37,6 +37,10 @@ void ShowWalletBackup() {
   NOTREACHED_IN_MIGRATION();
 }
 
+void UnlockWallet() {
+  NOTREACHED_IN_MIGRATION();
+}
```
Comment (supermassive) [inline browser/brave_wallet/brave_wallet_provider_delegate_impl_helper.cc:42]: NOTREACHED() both.

**Review by supermassive — APPROVED** [review]
wallet core lgtm with some suggestions

**Pull request merged**